lib/phoenix_live_view/diff.ex

```diff
@@ -341,12 +341,12 @@ defmodule Phoenix.LiveView.Diff do
   end
 
   defp configure_socket_for_component(socket, assigns, private, prints) do
-    %{
+    Utils.clear_flash(%{
       socket
       | assigns: assigns,
         private: private,
         fingerprints: prints
-    }
+    })
```

Comment:
`clear_flash` will mark flash as changed, which is probably not what we want and it may disable other optimizations (for example, checks if we should call render at all or not). I would do `Map.put(assigns, :flash, %{})`.